### ISSUE TITLE
Diagnostics don't seem to report file they're loaded from

### ISSUE THREAD

**Issue by Mythra** [google/starlark-rust#275]
Sorry for all the issue spam recently, thanks for the crate! Messing around with the error part of the starlark crate in order to provide a useful error message I've been looking at the span-traces natively provided by the starlark crate. Unfortunately it seems impossible to actually identify what file a particular issue is being caught in, and then where in that file it's being thrown.

To show this I've been messing with a simple example:

My code for what it's worth:

```rust
pub fn eval_config(mut root: PathBuf) -> Result<...> {
	let (mut global_env, mut type_values) = global_environment();
	pyra_module(&mut global_env, &mut type_values);
	global_env.freeze();
	let app_env = global_env.child("pyra");
	let map = Arc::new(Mutex::new(codemap::CodeMap::new()));
	let loader = SimpleFileLoader::new(&map, app_env);
	let load_res = loader.load(root.to_str().unwrap(), &type_values);
	if let Err(err) = load_res {
		return Err(eyre!("{:?}", err));
	}
        // ....
}
```

Root in this case is a file that looks like the following:

```starlark
load("./.github/other.star", "my_fun")
my_fun()
```

and other star looks like:

```starlark
def my_fun():
  pyra_host_executor(
    "a"
  )
```

This generates an error as `pyra_host_executor()` actually accepts no arguments. Unfortunately printing out the error (and looking at docs for maybe any non-printed values) doesn't show anyway to find that the file loading was in another file (and the column numbers are way too high):

```text
   0: DiagnosedError(Diagnostic { level: Error, message: "Extraneous parameter passed to function call", code: Some("CF05"), spans: [SpanLabel { span: Span { low: Pos(65), high: Pos(96) }, label: Some("Extraneous parameter in function call"), style: Primary }] })
```

Is there anyway to properly determine the file number + correct spans so i can print out a message to the user with where in the file the error is located?

**Comment by damienmg** [comment]
Files should be present in the created CodeMap. e.g. in https://github.com/google/starlark-rust/blob/master/starlark-repl/src/lib.rs#L82 we print the path along with the lines.

I think the spans are incorrect because they are representing a span in an abstract file.

**Comment by Mythra** [comment]
Hmmm that makes sense, I didn't see a way to get the file information, but I'll dig a bit more! I probably just missed something! I'll go ahead and close for now, and re-open if something comes up.

Thanks so much!
